ultra/ultra/adapters/default_reward_adapter.py

```diff
@@ -97,45 +99,48 @@ def adapt(observation: Observation, reward: float) -> float:
     #     ignore_vehicle_behind=True,
     # )
 
-    speed_fraction = max(0, ego_observation.speed / closest_wp.speed_limit)
+    # Environment reward from SMARTS
+    env_reward /= 100
+
+    # Termination reward
+    ego_reached_goal_reward = 1.0 if ego_events.reached_goal else 0.0
+
+    # Termination penalty
+    ego_collision_penalty = -1.0 if len(ego_events.collisions) > 0.0 else 0.0
+    ego_off_road_penalty = -1.0 if ego_events.off_road else 0.0
+    ego_not_moving_penalty = -1.0 if ego_events.not_moving else 0.0
+    ego_off_route_penalty = -1.0 if ego_events.off_route else 0.0
+    ego_wrong_way_penalty = -1.0 if ego_events.wrong_way else 0.0
+
+    # Intermediate rewards/penalties per step
     ego_step_reward = 0.02 * min(speed_fraction, 1) * np.cos(angle_error)
+    ego_dist_center_penalty = -0.005 * min(1, abs(ego_dist_center))
+    ego_angle_error_penalty = 0.005 * max(0, np.cos(angle_error))
+    ego_linear_jerk = -0.0001 * linear_jerk
+    ego_angular_jerk = -0.0001 * angular_jerk * math.cos(angle_error)
+
+    # Speed reward
     ego_speed_reward = min(
         0, (closest_wp.speed_limit - ego_observation.speed) * 0.01
     )  # m/s
-    ego_collision = len(ego_events.collisions) > 0
-    ego_collision_reward = -1.0 if ego_collision else 0.0
-    ego_off_road_reward = -1.0 if ego_events.off_road else 0.0
-    ego_off_route_reward = -1.0 if ego_events.off_route else 0.0
-    ego_wrong_way = -0.02 if ego_events.wrong_way else 0.0
-    ego_goal_reward = 0.0
-    ego_time_out = 0.0
-    ego_dist_center_reward = -0.002 * min(1, abs(ego_dist_center))
-    ego_angle_error_reward = -0.005 * max(0, np.cos(angle_error))
-    ego_reached_goal = 1.0 if ego_events.reached_goal else 0.0
+    ego_speed_reward += -0.01 if speed_fraction < 0.01 else 0.0  # Speed below threshold
+    ego_speed_reward = -0.1 if speed_fraction >= 1 else 0.0  # Speed above limit
```

Comment:
If `speed_fraction < 0.01`, the `ego_speed_reward` after line 127 will be 0 (and not the desired value of -0.01) since we assign it a value of 0 if `speed_fraction >= 1` in line 127.

Also, the original reward didn't take into account the `ego_speed_reward` calculation on line 123. I think it's okay if we do the same? What are your thoughts?

Reply:
resolved offline